File: 1TwoSum.cpp
#include<iostream>
#include<vector>
#include<unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        vector<int> result;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (m.find(complement) != m.end()) {
                result.push_back(i);
                result.push_back(m[complement]);
            } else {
                m.insert({nums[i], i});
            }
        }
        return result;
    }
};

int main () {
    Solution s;
    
    vector<int> numbers = {3,2,4};

    int target = 6;
    vector<int> answer = s.twoSum(numbers, target);

    for (auto i: answer) {
        cout << i << endl;
    }

}

File: 1TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        hashmap = {}
        for i,n in enumerate(nums):
            difference = target - n
            if difference in hashmap:
                return [hashmap[difference], i]
            hashmap[n] = i
            

if __name__ == "__main__":
    s = Solution();
    numbers = [1,2,3,4]
    target = 3

    answer = s.twoSum(numbers, target)

    for i in answer:
        print(i)

